engines/python/setup/djl_python/neuron_utils/model_loader.py

```diff
@@ -808,3 +808,115 @@ def partition(self, save_path: str, **kwargs):
+
+        on_dev_sampling_config = None
+        if self.config.on_device_generation:
+            on_dev_sampling_config = copy.deepcopy(self.model_config.generation_config)
+
+        quant_config = None
+        if self.config.neuron_quant:
+            quant_dtype = 's8'
+            dequant_dtype = 'bf16'
+            logging.info(f'Quantization is enabled. '
+                         f'quant_dtype: {quant_dtype}, '
+                         f'dequant_dtype: {dequant_dtype}')
+            quant_config = QuantizationConfig(
+                quant_dtype=quant_dtype,
+                dequant_dtype=dequant_dtype
+            )
+
+        weight_tiling = True
+        qkv_tiling = True
+        fuse_mlp = False
+        mlp_out_weight_transpose = False
+
+        if self.config.multi_node:
+            weight_tiling = False
+            qkv_tiling = False
+            fuse_mlp = True
+            mlp_out_weight_transpose = True
+
+        if self.config.speculative_draft_model:
+            weight_tiling = False
+            qkv_tiling = False
+            fuse_mlp = False
+            mlp_out_weight_transpose = False
+
+        neuron_config = NeuronConfig(
+            continuous_batching=continuous_batching_config,
+            attention_layout="BSH",
+            on_device_embedding=self.config.on_device_embedding,
+            on_device_generation=on_dev_sampling_config,
+            collectives_layout="BSH",
+            quant=quant_config,
+            fuse_qkv=True,
+            compilation_worker_count=self.config.compilation_worker_count,
+            sequence_parallel_norm=self.config.sequence_parallel,
+            sequence_parallel_norm_threshold=1,
+            shard_over_sequence=self.config.shard_over_sequence,
+            weight_tiling=weight_tiling,
+            qkv_tiling=qkv_tiling,
+            fuse_mlp=fuse_mlp,
+            mlp_out_weight_transpose=mlp_out_weight_transpose,
+        )
+        logging.info(f'Neuron config: {neuron_config}')
+        logging.info(f'neuron_config.on_device_embedding: {neuron_config.on_device_embedding}')
+        logging.info(f'neuron_config.on_device_generation: {neuron_config.on_device_generation}')
+        logging.info(f'neuron_config.shard_over_sequence: {neuron_config.shard_over_sequence}')
+        logging.info(f'neuron_config.continuous_batching_config: {neuron_config.continuous_batching}')
+        logging.info(f'neuron_config.neuron_cc_pipeline_factor: {self.config.neuron_cc_pipeline_factor}')
+        logging.info(f'batch_size: {self.config.max_rolling_batch_size}')
+        return neuron_config
+
+    def load_model(self, **kwargs):
+        from vllm.model_executor.model_loader.neuron import get_neuron_sd_model
+        logging.info(f'context_length_estimate: {self.config.context_length_estimate}')
+        logging.info(f"Amp {self.config.amp}")
+        if self.config.context_length_estimate is None:
```

Comment:
Use defaults from tnx model loader - maybe turn it into a static method in utils - and take the smart default or the config if it is provided

Reply:
Done